dkpro-core-io-bincas-asl/src/main/java/de/tudarmstadt/ukp/dkpro/core/io/bincas/BinaryCasReader.java

```
@@ -133,6 +171,44 @@ public void getNext(CAS aCAS)
         }
     }
     
+    // Legacy DKPro file format support
+ 	private TypeSystemImpl readDKProHeader(BufferedInputStream bis, byte[] header,
+ 			TypeSystemImpl ts) {
+ 		// Check if this is original UIMA CAS format or DKPro Core format
+ 		if (Arrays.equals(header, DKPRO_HEADER)) {
+ 			// If it is DKPro Core format, read the type system
+ 			getLogger().debug("Found DKPro-Core-style embedded type system");
+ 			ObjectInputStream ois;
+ 			try {
+ 				ois = new ObjectInputStream(bis);
+ 				CASMgrSerializer casMgr = (CASMgrSerializer) ois.readObject();
+ 				if (ts == null) {
+ 					ts = casMgr.getTypeSystem();
+ 					ts.commit();
+ 				}
+ 			} catch (IOException e) {
+ 				e.printStackTrace();
+ 			} catch (ClassNotFoundException e) {
+ 				e.printStackTrace();
+ 			}
```

catch (IOException | ClassNotFoundException e) {
  throw new CollectionException(e);
}

Please never "swallow" exceptions via catch/e.printStackTrace.